core/src/main/java/org/apache/accumulo/core/file/FilePrefix.java

```diff
@@ -29,18 +31,8 @@ public enum FilePrefix {
   }
 
   public static FilePrefix fromPrefix(String prefix) {
-    switch (prefix) {
-      case "I":
-        return BULK_IMPORT;
-      case "F":
-        return MINOR_COMPACTION;
-      case "C":
-        return MAJOR_COMPACTION;
-      case "A":
-        return MAJOR_COMPACTION_ALL_FILES;
-      default:
-        throw new IllegalArgumentException("Unknown prefix type: " + prefix);
-    }
+    return Stream.of(FilePrefix.values()).filter(p -> p.prefix.equals(prefix)).findAny()
+        .orElseThrow(() -> new IllegalArgumentException("Unknown prefix type: " + prefix));
```

Comment:
When I proposed this on #3148, I did not test it. The FilePrefix class should have a unit test that verifies this works correctly.